pkg/agent/apiserver/handlers/memberlist/handler.go

```diff
@@ -48,9 +48,14 @@ func generateResponse(node *v1.Node, aliveNodes sets.String) Response {
 // HandleFunc returns the function which can handle queries issued by the memberlist command.
 func HandleFunc(aq querier.AgentQuerier) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		memberlistCluster := aq.GetMemberlistCluster()
+		if memberlistCluster == nil {
```

Comment:
An interface assigned with a concrete type's value will not be `nil` regardless whether the concrete type is nil or not.
Try this code: https://go.dev/play/p/rkyXVVwDjw1.

The nil check should use `reflect.ValueOf(multiclusterInterface).IsNil()`